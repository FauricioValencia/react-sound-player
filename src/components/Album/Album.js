import React from 'react'
import PropTypes from 'prop-types'

import ErrorBoundary from 'HOCS/ErrorBoundary'
import { GlobalContext } from 'store/GlobalStore'

const Album = ({ className }) => {
  const [store] = React.useContext(GlobalContext)

  return (
    <ErrorBoundary>
      <div id="album-art" className={className}>
        <img
          src={store?.song?.songsImage}
          className="active"
          id={store?.song?.albumArtwork}
          alt={store?.song?.trackName}
        />
        <div id="buffer-box">Buffering ...</div>
      </div>
    </ErrorBoundary>
  )
}

Album.propTypes = {
  className: PropTypes.string,
  song: PropTypes.object,
}

export default Album
