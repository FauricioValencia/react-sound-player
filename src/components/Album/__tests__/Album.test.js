import { shallow } from 'enzyme'

import Album from '../Album'

const testProps = { className: 'active', song: {} }

describe('Album component', () => {
  describe('GIVEN testProps', () => {
    describe('WHEN the component is mounted', () => {
      const wrapper = shallow(<Album {...testProps} />)
      it('THEN should show correctly', () => {
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})
