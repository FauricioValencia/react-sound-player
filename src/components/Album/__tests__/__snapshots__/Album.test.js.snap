// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album component GIVEN testProps WHEN the component is mounted THEN should show correctly 1`] = `ShallowWrapper {}`;
